### DISCUSSION TITLE
Use Bun for dependency management

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/indexers#51] [Ideas]
Refs:

- https://github.com/sablier-labs/v2-core/issues/759
- https://github.com/sablier-labs/sablier-v2-integration-template/pull/3

**Comment by razgraf** [comment]
This isn't possible at the moment as the repository currently uses two package managers: Envio needs `pnpm`, the subgraphs are built using `yarn`.

I've tried updating to `pnpm` at some point but the subgraph projects weren't very happy about it (something to do with the extended `tsconfig`).

As soon as Envio starts supporting `bun` we can circle back to this issue. It's important we use only one package manager so we can refactor the codebase to promote re-usability between subgraphs and indexers.

**Reply by PaulRBerg** [reply]
Got it. Did you check if Envio supports `bun` as of lately?

**Reply by PaulRBerg** [reply]
I was following the docs and have found it quite confusing/ bad DX to have to use both yarn and pnpm.

**Reply by razgraf** [reply]
Envio only works with `pnpm`, yes.

They haven't updated this anyway, but I did ask a while ago about it so they're aware it's something we would like to see.

**Comment by PaulRBerg** [comment]
I was able to make it work with Bun, but Envio will still forcefully use PNPM over our package manager when `envio codegen` is run.

Related: https://github.com/enviodev/docs/issues/622

It remains to be seen if we go with Bun or PNPM. I am working on the Envio indexers now.

**Comment by PaulRBerg** [comment]
Ended up switching to PNPM because Envio is force-feeding it to us

https://github.com/sablier-labs/indexers/pull/92
